lib/api/core/models/repositories/roleRepository.js

```diff
@@ -29,30 +29,39 @@ module.exports = function (kuzzle) {
       deferred = q.defer(),
       keys = [],
       buffer = {},
-      result = [];
+      result = [],
+      internalIndexAllowedIds = {},
+      restrictions = {};
 
-    roleKeys.forEach((roleKey) => {
+    roleKeys.forEach(roleKey => {
```

Comment:
@ballinette if you could adapt your PR to the actual code it would be deeply appreciated